### ISSUE TITLE
Add option to stop a gulp task

### ISSUE THREAD

**Issue by mklakakilli** [jacogr/atom-gulp-control#6]
I have a gulp task called `serve` which turns on the server for my application at specific port. I want to be able to stop this task at my will.

**Comment by robertbaker** [comment]
+1

**Comment by andrew-makarenko** [comment]
+1

**Comment by bhsnll** [comment]
+1

**Comment by Zoltu** [comment]
Gulp task will only run one task at a time so you can workaround this by creating a "stop" task that does nothing.  Personally, when I want to stop serving I just run my build or clean task.

**Comment by dancvogel** [comment]
+1. I have the same issue. If I run a different task, close the gulp-control tab, or even close Atom there are still python.exe processes running my server. I didn't realize this and one day I found I had 30 python processes running!

**Comment by JakeCobley** [comment]
+1

**Comment by ericwooley** [comment]
+1

**Comment by NicolasPio** [comment]
:+1:

**Comment by nlenkowski** [comment]
Agreed, it would be very nice to be able to stop watch tasks and destroy the process. Any progress on this?

**Comment by acburdine** [comment]
Did some digging into this problem behind this. It seems to be related to the fact that atom's BufferedProcess class doesn't allow for sending a code via process.kill(). Therefore, child processes aren't killed correctly.

Relevant issue (i think): https://github.com/atom/atom/issues/7252

**Comment by cbobbett** [comment]
+1

**Comment by munkacsimark** [comment]
👍

**Comment by tdietrich513** [comment]
+1, created a stop task but this should really be a built in feature.

**Comment by acburdine** [comment]
This should be fixed once a release is made :smile:

**Comment by xgvargas** [comment]
:+1:

**Comment by rockResolve** [comment]
+1
